File: CCC/bridgeTransport.cpp
#include <iostream>

using namespace std;

int main(){
    int w,n;
    cin>>w;
    cin>>n;
    int carts[n];
    for (int i = 0; i < n; i++)
    {
        cin>>carts[i];
    }

    int totalMass = 0;
    int cartsCounted=0;

    for (int i = 0; i < n; i++)
    {
        totalMass += carts[i];
        
        if (totalMass>w)
        {
            break;
        }
        cartsCounted++;
        if (cartsCounted>=4)
        {
            totalMass-=carts[cartsCounted-4];
        } 
    }

    cout<<cartsCounted<<endl;
    

}

File: CCC/bridgeTransport.py
w = int(input())
n = int(input())
carts = []
for i in range(n):
    cart = int(input())
    carts.append(cart)

cartsCounted = 0
mass = 0

for i in range(n):
    mass+= carts[i]
    if mass > w:
        break
    cartsCounted+=1
    if cartsCounted>=4:
        mass-=carts[i-3]
print(cartsCounted)